src/test/java/org/springframework/data/couchbase/core/CouchbaseTemplateQueryIntegrationTests.java

```diff
@@ -148,6 +156,55 @@ void findByMatchingQuery() {
 		assertEquals(1, foundUsers.size());
 	}
 
+	@Test
+	void findByMatchingQueryProjected() {
+
+		UserSubmission user = new UserSubmission();
+		user.setId(UUID.randomUUID().toString());
+		user.setUsername("dave");
+		user.setRoles(Arrays.asList("role1", "role2"));
+		Address address = new Address();
+		address.setStreet("1234 Olcott Street");
+		user.setAddress(address);
+		user.setSubmissions(
+				Arrays.asList(new Submission(UUID.randomUUID().toString(), user.getId(), "tid", "status", 123)));
+		user.setCourses(Arrays.asList(new Course(UUID.randomUUID().toString(), user.getId(), "581"),
+				new Course(UUID.randomUUID().toString(), user.getId(), "777")));
+		couchbaseTemplate.upsertById(UserSubmission.class).one(user);
+
+		Query daveUsers = new Query(QueryCriteria.where("username").like("dave"));
+
+		final List<UserSubmissionProjected> foundUserSubmissions = couchbaseTemplate.findByQuery(UserSubmission.class)
+				.as(UserSubmissionProjected.class).consistentWith(QueryScanConsistency.REQUEST_PLUS).matching(daveUsers).all();
+		assertEquals(1, foundUserSubmissions.size());
+		assertEquals(user.getUsername(), foundUserSubmissions.get(0).getUsername());
+		assertEquals(user.getId(), foundUserSubmissions.get(0).getId());
+		// assertEquals( user.getSubmissions(), foundUserSubmissions.get(0).getSubmissions());
+		assertEquals(user.getCourses(), foundUserSubmissions.get(0).getCourses());
+		assertEquals(user.getAddress(), foundUserSubmissions.get(0).getAddress());
+
+		System.out.println(foundUserSubmissions.get(0));
+
+		couchbaseTemplate.removeByQuery(UserSubmission.class).all();
+
+		User user1 = new User(UUID.randomUUID().toString(), "user1", "user1");
+		User user2 = new User(UUID.randomUUID().toString(), "user2", "user2");
+		User specialUser = new User(UUID.randomUUID().toString(), "special", "special");
+
+		couchbaseTemplate.upsertById(User.class).all(Arrays.asList(user1, user2, specialUser));
+
+		Query specialUsers = new Query(QueryCriteria.where("firstname").like("special"));
+		final List<UserJustLastName> foundUsers = couchbaseTemplate.findByQuery(User.class).as(UserJustLastName.class)
+				.consistentWith(QueryScanConsistency.REQUEST_PLUS).matching(specialUsers).all();
+		assertEquals(1, foundUsers.size());
+
+		final List<UserJustLastName> foundUsersReactive = reactiveCouchbaseTemplate.findByQuery(User.class)
```

Comment:
Midway in review I have a basic question.. so here we have:

```
reactiveCouchbaseTemplate.findByQuery(User.class).as(UserJustLastName.class)
```

what is the reason we could not just do 

reactiveCouchbaseTemplate.findByQuery(UserJustLastName.class)

I assume there is good reason, but I probably don't understand it fully

Reply:
Good question. Because the documents are stored with _class = "somepackage.User" and they would not be selected by the query.  This is also the reason that there is no as(otherClass) method for ExecutableFindById.